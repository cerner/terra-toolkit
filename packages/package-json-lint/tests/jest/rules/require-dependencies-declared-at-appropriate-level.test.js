const requireDependenciesDeclaredAtAppropriateLevel = require('../../../src/rules/require-dependencies-declared-at-appropriate-level');

describe('require-dependencies-declared-at-appropriate-level', () => {
  describe('when projectType is application', () => {
    it('fails as a warning when there unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'application',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        '@cerner/webpack-config-terra': '^2.0.0',
      });
      rule.peerDependencies({
        jest: '^26.6.2',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there are no unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'application',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        classnames: '^2.2.5',
      });
      rule.peerDependencies({
        react: '^16.8.5',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when unnecessary dependencies/peerDependencies are passed in the allowList', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
            allowList: ['@cerner/webpack-config-terra', 'jest'],
          },
        },
        projectType: 'application',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        '@cerner/webpack-config-terra': '^2.0.0',
      });
      rule.peerDependencies({
        jest: '^26.6.2',
      });
      expect(results).toMatchSnapshot();
    });
  });

  describe('when projectType is module', () => {
    it('fails as a warning when there unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'module',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        webpack: '^4.30.0',
      });
      rule.peerDependencies({
        jest: '^26.6.2',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there are no unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'module',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        classnames: '^2.2.5',
      });
      rule.peerDependencies({
        react: '^16.8.5',
      });
      expect(results).toMatchSnapshot();
    });
  });

  describe('when projectType is devModule', () => {
    it('succeeds when there unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'devModule',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        '@cerner/webpack-config-terra': '^2.0.0',
      });
      rule.peerDependencies({
        jest: '^26.6.2',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there no unnecessary dependencies/peerDependencies', () => {
      const results = [];
      const rule = requireDependenciesDeclaredAtAppropriateLevel.create({
        ruleConfig: {
          severity: {
            severityType: 'warn',
          },
        },
        projectType: 'devModule',
        report: (issues) => {
          results.push(issues);
        },
      });
      rule.dependencies({
        classnames: '^2.2.5',
      });
      rule.peerDependencies({
        react: '^16.8.5',
      });
      expect(results).toMatchSnapshot();
    });
  });
});
