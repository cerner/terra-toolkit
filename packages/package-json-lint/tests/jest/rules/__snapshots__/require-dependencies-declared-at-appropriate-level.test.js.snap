// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-dependencies-declared-at-appropriate-level when projectType is application fails as a warning when there unnecessary dependencies/peerDependencies 1`] = `
Array [
  Object {
    "lintId": "require-dependencies-declared-at-appropriate-level",
    "lintMessage": "This project has unnecessary dependencies that violates the require-dependencies-declared-at-appropriate-level rule:
  @cerner/webpack-config-terra@^2.0.0 does not satisfy requirement for require-dependencies-declared-at-appropriate-level rule.",
    "projectType": "application",
    "severity": Object {
      "severityType": "warn",
    },
  },
  Object {
    "lintId": "require-dependencies-declared-at-appropriate-level",
    "lintMessage": "This project has unnecessary peerDependencies that violates the require-dependencies-declared-at-appropriate-level rule:
  jest@^26.6.2 does not satisfy requirement for require-dependencies-declared-at-appropriate-level rule.",
    "projectType": "application",
    "severity": Object {
      "severityType": "warn",
    },
  },
]
`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is application succeeds when there are no unnecessary dependencies/peerDependencies 1`] = `Array []`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is application succeeds when unnecessary dependencies/peerDependencies are passed in the allowList 1`] = `Array []`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is devModule succeeds when there no unnecessary dependencies/peerDependencies 1`] = `Array []`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is devModule succeeds when there unnecessary dependencies/peerDependencies 1`] = `Array []`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is module fails as a warning when there unnecessary dependencies/peerDependencies 1`] = `
Array [
  Object {
    "lintId": "require-dependencies-declared-at-appropriate-level",
    "lintMessage": "This project has unnecessary peerDependencies that violates the require-dependencies-declared-at-appropriate-level rule:
  jest@^26.6.2 does not satisfy requirement for require-dependencies-declared-at-appropriate-level rule.",
    "projectType": "module",
    "severity": Object {
      "severityType": "warn",
    },
  },
]
`;

exports[`require-dependencies-declared-at-appropriate-level when projectType is module succeeds when there are no unnecessary dependencies/peerDependencies 1`] = `Array []`;
